Makefile

```
@@ -190,3 +190,11 @@ check_autosquash: check_env
 readme: check_docker
 	docker pull evkalinin/gh-md-toc:0.7.0
 	docker run -it -v $$PWD:/build evkalinin/gh-md-toc:0.7.0 --no-backup /build/README.md
+
+.PHONY: tdr_canned_bundle
+tdr_canned_bundle: check_python
```

Is the point of this target just to make sure that the bundles are always generated in the same order? When is the target supposed to run?

If someone calls the script manually, instead of through the Makfile, you'll have a duplicate UUID, right? That would probably cause problems. If this is the case, consider putting a note in the script that warns it should only be called via the Makefile.

Finally, consider
```suggestion
tdr_canned_bundles: check_python
```

---

> Is the point of this target just to make sure that the bundles are always generated in the same order?

Only bundle is generated. The point is to avoid the tedium/potential source of error of having to manually match the snapshot and bundle UUID to their values in `test_tdr.py`, as I frequently had to do while testing my changes. My other changes in that commit ensure that the _entities_ are consistently sorted within the canned bundle, which helps keep the diff small when changes to the recanning script are made, as in the second commit.

> When is the target supposed to run?

When changes are made to the recanning script that affect its output.

> If someone calls the script manually, instead of through the Makfile, you'll have a duplicate UUID, right?

I'm not entirely sure what you mean by this, but I think you're referring to the eventuality of generating a second canned bundle from a different underlying DSS bundle, thus resulting in two canned TDR bundles where different entities have the same UUID? I hadn't considered that, but I can see how it could be a potential problem. I don't see what it has to do with the make target though. It strikes me as a consequence of making the generated UUIDs deterministic; the duplicates would occur whenever running the script with multiple bundles, regardless of whether the script is invoked through the make target or not.

Have I correctly interpreted your concern?

> Finally, consider ...

The script only outputs a single bundle per invocation, and the target only invokes the script once. I believe the target is correctly named. I think using a plural here would lead to confusion regarding the format of the script's output.

---

Assuming you meant
> Only **one** bundle is generated
I overlooked that, sorry. The name definitely makes sense as is.

Thanks for the thorough explanation in regards to everything else.

> Have I correctly interpreted your concern?

Yes.

> the duplicates would occur whenever running the script with multiple bundles, regardless of whether the script is invoked through the make target or not.

You're correct, I was mistaken. I assumed the script could (and was) operating on multiple bundles.

We agree though, if the script is run multiple times on different bundles, the outputs will share "random" UUIDs. If you think that could lead to some unexpected bugs (perhaps if both bundles are used in the same test) then consider either fixing the behavior, or document this in the `recan_bundle_tdr.py` script.

---

Yeah, having "UUID"s that aren't unique feels like an accident waiting to happen. I think I can easily fix this by using the bundle UUID as the namespace instead of hardcoding an arbitrary value.

Oh, and I was mistaken about my changes causing the entities to be sorted, I was thinking of a different PR. The ordering is deterministic, not sorted.